pkg/go/graph/weighted_graph_edge.go

```diff
@@ -0,0 +1,75 @@
+package graph
+
+import (
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/graph/encoding"
+)
+
+type WeightedAuthorizationModelEdge struct {
+	*AuthorizationModelEdge
+	weights WeightMap
+
+	// isNested signals that this edge is a self-loop. The associated node will also have this flag set to true.
+	isNested bool
```

Comment:
Put in other words, is this an edge that connects a vertex to itself?
```mermaid
  graph TD;
      A-->A;
```

Reply:
Yep! not sure what to call it